### PR TITLE
recreate inline policy document for iam roles if policy changes

### PR DESCRIPTION
**Pull request by bdehri**
Towards https://github.com/giantswarm/giantswarm/issues/29985

### CODE REVIEW

File: pkg/iam/iam.go [hunk 1]
```diff
@@ -361,44 +351,45 @@ func (s *IAMService) attachInlinePolicy(roleName string, roleType string, params
...
 
-	alreadyExists := false
-
 	// check if the inline policy already exists
 	o, err := s.iamClient.ListRolePolicies(i)
 	if err == nil {
 		for _, p := range o.PolicyNames {
 			if *p == policyName(s.roleType, s.clusterName) {
-				alreadyExists = true
+				_, err := s.iamClient.DeleteRolePolicy(&awsiam.DeleteRolePolicyInput{
```
Comment (calvix) [inline pkg/iam/iam.go:359]: is there a way how to check if there is a diff that you need the deletion/update instead of doing it every time?

Comment (bdehri) [reply pkg/iam/iam.go:359]: We can use `GetRolePolicy` to read policy and do json compare somehow.

File: pkg/iam/iam.go [hunk 2]
```diff
@@ -357,49 +350,60 @@ func (s *IAMService) applyAssumePolicyRole(roleName string, roleType string, par
...
 	}
 
-	// add inline policy to the main IAM role if it do not exist yet
-	if !alreadyExists {
-		tmpl := getInlinePolicyTemplate(roleType)
-
-		policyDocument, err := generatePolicyDocument(tmpl, params)
+	if err == nil {
+		l.Info("output policy document", "policy_document", *output.PolicyDocument)
+		l.Info("new policy document", "policy_document", policyDocument)
```
Comment (calvix) [inline pkg/iam/iam.go:373]: I think this might be too verbose for normal logs, should be hidden under Debug or something

**Review by calvix — APPROVED** [review]
lgtm

**Pull request merged**